### ISSUE TITLE
[Feature Request] Mega Widget

### ISSUE THREAD

**Issue by netsonicyxf** [WangDaYeeeeee/GeometricWeather#380]
[Weather Mate (Weather M8)](https://play.google.com/store/apps/details?id=pro.burgerz.miweather8&hl=zh&gl=US) has a very powerful widget:
1. It has two lines, the top line shows hour forcast, while the bottom line shows week forcast.
![image](https://user-images.githubusercontent.com/43513391/180631282-c52e8cf0-af06-4797-b35a-7cfe477e381a.png)
2. When clicking the hour forcast, it switches bwtween other weather info
![image](https://user-images.githubusercontent.com/43513391/180631326-a8129dda-8d27-4cb1-9058-5d9d6856c595.png)
![image](https://user-images.githubusercontent.com/43513391/180631332-ed3d9d99-06e9-41b3-9fb3-ac11d2702dba.png)
3. It can optionally shows when the weather is updated, 
![image](https://user-images.githubusercontent.com/43513391/180631438-77b22d25-fa6c-4c64-8790-0f7e27de571e.png)
and shows which weather source is used
![image](https://user-images.githubusercontent.com/43513391/180631449-18036b0c-e024-441c-9c5f-9373f3000404.png)

PLease make a similar widge for [GeometricWeather](https://github.com/WangDaYeeeeee/GeometricWeather)
I have other suggestion: Please adding useful infomations at the title of the widget, like this
Title: Location, Precipitation (optional), Air Quality (optional), UV radiation (optional), weather updated time (optional)
Top line: hour forcast
Bottom line: week forest
Bottom: weather source (optional)
